xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentsinline.vm

```diff
@@ -256,10 +250,20 @@ $xwiki.ssfx.use('uicomponents/viewers/comments.css', true)
 ##
 ##
 #macro(displayCommentButton)
-<button type="button" id="openCommentForm" class="btn btn-default">
-  $services.icon.renderHTML('comment')
-  $services.localization.render('core.viewers.comments.add.comment.label')
-</button>
+  <button type="button" id="openCommentForm" class="btn btn-default">
+    $services.icon.renderHTML('comment')
+    #if($xcontext.user != 'XWiki.XWikiGuest')
+      $services.localization.render('core.viewers.comments.add.comment.label')
+    #else
+      $services.localization.render('core.viewers.comments.add.commentAsGuest.label')
+    #end
+  </button>
+  #if($xcontext.user == 'XWiki.XWikiGuest')
```

Comment:
I think you could use directly `$isGuest`

Reply:
This boolean is used in multiple other .vm files (e.g. [login.vm](https://github.com/xwiki/xwiki-platform/blob/ddc61325ba827f194bd1eb31df8bbf63adfef81f/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/login.vm#L20)).
Let me know if I should open a new ticket to correct them.
I fixed the ones in this file in https://github.com/xwiki/xwiki-platform/pull/2769/commits/6e8dc2bd5efd90cb98e027aac256c6cdd297e3f2 :+1: .